enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+
+A separate effort is underway to implement Centralized Machine Management,
+where Machine API providers and related resources would all be implemented on
+the hub cluster instead of the spoke cluster. Rather than use a MachinePool on
+the hub as a proxy for a remote MachineSet resource, the entire Machine API
+affecting a spoke cluster would be available for use on the Hub. That will be
+the starting point for integrating agent-based installation with Machines.
+
+Typically with the Machine API, a MachineSet gets scaled up, one or more
+Machines get created, and then a controller (part of a particular platform's
+machine-api-provider) interacts with a provider API (such as a public cloud
+API) to create or obtain a host per Machine that can be provisioned as Nodes in
+the cluster.
+
+Integrating agent-based provisioning into that flow will involve matching each
+Machine resource with an Agent resource in a similar way to how the bare metal
+provider currently matches a Machine with an availbale BareMetalHost. On a bare
+metal platform, Agents will likely continue to be created in advance,
+independently of a Machine. It is possible that other platforms may create
+Agents on-demand when a Machine appears.
+
+#### Hive Cluster Adoption
+
+Hive is able to adopt existing clusters that it did not create. This is done by
+creating a ClusterDeployment where `spec.installed == true` and providing a
+kubeconfig.
+
+In order to adopt a cluster and then add workers using an agent-based approach,
+the user must additionally:
+* specify the agent label selector on the ClusterDeployment's Spec
+* add a reference to the ClusterDeployment on one or more InstallEnvs
+
+Otherwise there are no differences in the InstallEnv or the workflow as
+described above.
+
+### Risks and Mitigations
+
+#### API Versions and Potential Change
+
+Agent-based installation is a significant feature set with significant API
+surface area. There is a strong potential for the need to change these new APIs
+after receiving feedback from users. However there is a competing desire to
+support these APIs when they are delivered in the OpenShift 4.8 timeframe,
+which rules out the use of an "alpha" API version due to its insufficient level
+of commitment.
+
+New CRDs will have version "v1beta1" during the OpenShift 4.8 timeline (though
+they will be distributed as part of RHACM). This will enable them to be
+supported in production while allowing greater flexibility for changes than if
+they were released as "v1".
+
+The agent-specific portions of the hive ClusterDeployment will inherit the
+ClusterDeployment's current version of "v1".
+
+#### Agent REST API Auth
+
+Agents will communicate with the assisted service using the non-kubernetes REST
+API that is currently in use by the SaaS. The backend assisted service will be
+listening on a socket that is routable from anywhere Agents are expected to
+run, so it needs to be secured from other potential actors on the network.
```

Comment:
Do we expect to need special networking for that, or would we use a Service to expose the endpoint?

Reply:
We'll use a Route to expose it.